### PR TITLE
Adds SAMsum, Q-Msum and SUMMScreen dataset

### PR DESCRIPTION
**Pull request by MuroriM**
Add SAMsum dataset:

- Adds the relevant changes to huggingface_datasets.py and __init__.py to download and load SAMsum

Does minor edits to fix datasets naming convention on Scisummnet

- Modifies: non_huggingface_datasets.py

### CODE REVIEW

File: dataset/huggingface_datasets.py [hunk 1]
```diff
@@ -109,3 +109,39 @@ def process_multinews_data(data: Dataset) -> List[SummInstance]:
...
+                         is_dialogue_based=True,
+                         is_multi_document=False,
+                         train_set=processed_train_set,
+                         dev_set=processed_dev_set,
+                         test_set=processed_test_set)
+        
+    @staticmethod
+    def process_samsum_data(data: Dataset) -> List[SummInstance]:
+        for instance in tqdm(data):
+            dialogue: str = instance['dialogue']
```
Comment (niansong1996) [inline dataset/huggingface_datasets.py:142]: This doesn't look right, dialogues should be in the form of `List[str]` where each string looks like `f"{speaker} : {utterance}"`

**Review by niansong1996 — COMMENTED** [review]
There is a comment about the type for dialogues, let me know if you have any questions.

**Comment by MuroriM** [comment]
I've fixed how the dialogue instances are stored. Each source instance is now a list of summ_instances containing strings in the form of '["speaker1 : utter..", "speaker2 : utter..."]

**Comment by niansong1996** [comment]
LGTM. @MuroriM  Can you confirm that the tests are passing?

**Comment by MuroriM** [comment]
It's passing the tests. I also modified to check for list[str] instances on source if it's a dialogue dataset

**Comment by niansong1996** [comment]
Awesome, merging this now

**Pull request merged**